packages/list-store/src/search-params.ts

```diff
@@ -3,53 +3,82 @@ import {SortBy} from './sort';
 
 export type Type = 'string' | 'array' | 'number';
 
+const defaultLimit = 25;
+
 // Only strings are allowed in the search params.
 const searchParamFilterSchema = z
-  .array(z.string())
-  .or(z.array(z.number().transform(value => `${value}`)))
-  .or(z.string())
-  .or(z.number().transform(value => `${value}`))
+  .array(z.coerce.string())
+  .or(z.coerce.string())
   .default('');
 
-function getSearchParamsSchema(defaultSortBy: string) {
+interface GetSearchParamsSchemaProps {
+  defaultSortBy: string;
+  defaultView?: string;
+  defaultImageFetchMode?: string;
+}
+
+function transformToStringAndRemoveDefault(defaultValue?: string) {
```

Comment:
The name, to me, suggests that this function actually transforms a value, but I think that it returns a Zod schema that we can then use for transforming a value. Perhaps rename to e.g. `transformToStringAndRemoveDefaultSchema`?